src/poliastro/atmosphere/models.py

```diff
@@ -0,0 +1,661 @@
+""" Holds the Atmosphere class. """
+
+import numpy as np
+from astropy import units as u
+
+from poliastro.atmosphere.util import (
+    R_air,
+    geometric_to_geopotential,
+    geopotential_to_geometric,
+    gravity,
+)
+from poliastro.constants import R_earth
+
+
+class COESA(object):
+    """ Class for U.S Standard Atmosphere models. """
+
+    def __init__(self, table):
+        """ Constructor for Atmosphere instances.
+
+        Parameters
+        ----------
+        table: list 1x5
+            List containing [Zb_table, Hb_table, Lb_table Tb_table, pb_table]
+        """
+
+        self._Zb_table = table[0]
+        self._Hb_table = table[1]
+        self._Tb_table = table[2]
+        self._Lb_table = table[3]
+        self._pb_table = table[4]
+
+    def _get_heights(self, alt, geometric=True):
+        """ Returns geometric and geopotential heights being passed altitude.
+
+        Parameters
+        ----------
+        alt: ~astropy.units.Quantity
+            Geometric/Geopotential height.
+        geometric: boolean
+            If `True`, assumes that `alt` argument is geometric kind.
+
+        Returns
+        -------
+        Z: ~ astropy.units.Quantity
+            Geometrical height.
+        H: ~astropy.units.Quantity
+            Geopotential height.
+        """
+
+        if geometric and (alt < self._Zb_table[0] or alt > self._Zb_table[-1]):
+            raise ValueError(
+                "Geometric altitude must be in range [{}, {}].".format(
+                    self._Zb_table[0], self._Zb_table[-1]
+                )
+            )
+
+        if not geometric and (alt < self._Hb_table[0] or alt > self._Hb_table[-1]):
+            raise ValueError(
+                "Geopotential altitude must be in range [{}, {}].".format(
+                    self._Hb_table[0], self._Hb_table[-1]
+                )
+            )
+
+        # Always work with geopotential height
+        if geometric and alt in self._Zb_table:
+            i = np.where(self._Zb_table == alt)
+            Z = self._Zb_table[i]
+            H = self._Hb_table[i]
+        elif geometric:
+            Z = alt
+            H = geometric_to_geopotential(alt, R_earth)
+        else:
+            Z = geopotential_to_geometric(alt, R_earth)
+            H = alt
+
+        return Z, H
+
+    def _get_base_index(self, H):
+        """ Gets corresponding base coefficient for given geopotential altitude.
+
+        Parameters
+        ----------
+        H: ~astropy.units.Quantity
+            Geopotential altitude.
+
+        Returns
+        -------
+        i: int
+            Index position in table.
+        """
+
+        for i, Hb in enumerate(self._Hb_table):
```

Comment:
Isn't this duplicated in `util.get_base_index`?

Reply:
Ups! You are right, I'll get rid of this function in the `util` script :+1: